src/api/Controllers/CoursesController.cs

```diff
@@ -223,6 +224,35 @@ private bool CircuitBreakerTripped(ICollection<Course> receivedCourses)
                 _logger.LogWarning( $"CircuitBreaker: bypassing checks as no limit configured. Configure with: {limitKey}");
                 return false;
             }
+
+            var courseInfoFormat = "[{0}, {1}]";
```

Comment:
pull more out into a small method instead of just the string

```
formatCourseInfo(provider,code){
 string.Format("[{0}, {1}]", course.Provider != null ? course.Provider.ProviderCode : null, course.ProgrammeCode))
}
```

splitting the string from the params breaks the IDE ability to validate them

also use the new interpolation `${thing}`